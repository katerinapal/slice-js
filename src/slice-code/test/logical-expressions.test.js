import {runAllCombosTests, comboOfBools} from './helpers/utils'
// import combs from 'combs'

// fit(
//   'ifElse(true)',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/logical-expressions'), ({andOr}) => {
//     return andOr(true, true, true)
//   })
// )

runAllCombosTests({
  filename: require.resolve('./fixtures/logical-expressions'),
  methods: [
    {methodName: 'allAnd', explicitArgs: comboOfBools(3)},
    {methodName: 'allOr', explicitArgs: comboOfBools(3)},
    {methodName: 'andOr', explicitArgs: comboOfBools(3)},
    {methodName: 'orAndOr', explicitArgs: comboOfBools(4)},
    {methodName: 'andOrAnd', explicitArgs: comboOfBools(4)},
  ],
})
