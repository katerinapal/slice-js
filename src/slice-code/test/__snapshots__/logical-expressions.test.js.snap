exports[`test allAnd(false, false, false) 1`] = `"export const allAnd = a => a;"`;

exports[`test allAnd(false, false, true) 1`] = `"export const allAnd = a => a;"`;

exports[`test allAnd(false, true, false) 1`] = `"export const allAnd = a => a;"`;

exports[`test allAnd(false, true, true) 1`] = `"export const allAnd = a => a;"`;

exports[`test allAnd(true, false, false) 1`] = `"export const allAnd = (a, b) => a && b;"`;

exports[`test allAnd(true, false, true) 1`] = `"export const allAnd = (a, b) => a && b;"`;

exports[`test allAnd(true, true, false) 1`] = `"export const allAnd = (a, b, c) => a && b && c;"`;

exports[`test allAnd(true, true, true) 1`] = `"export const allAnd = (a, b, c) => a && b && c;"`;

exports[`test allOr(false, false, false) 1`] = `
"
export const allOr = (a, b, c) => a || b || c;"
`;

exports[`test allOr(false, false, true) 1`] = `
"
export const allOr = (a, b, c) => a || b || c;"
`;

exports[`test allOr(false, true, false) 1`] = `
"
export const allOr = (a, b) => a || b;"
`;

exports[`test allOr(false, true, true) 1`] = `
"
export const allOr = (a, b) => a || b;"
`;

exports[`test allOr(true, false, false) 1`] = `
"
export const allOr = a => a;"
`;

exports[`test allOr(true, false, true) 1`] = `
"
export const allOr = a => a;"
`;

exports[`test allOr(true, true, false) 1`] = `
"
export const allOr = a => a;"
`;

exports[`test allOr(true, true, true) 1`] = `
"
export const allOr = a => a;"
`;

exports[`test andOr(false, false, false) 1`] = `
"
export const andOr = (a, b, c) => a || c;"
`;

exports[`test andOr(false, false, true) 1`] = `
"
export const andOr = (a, b, c) => a || c;"
`;

exports[`test andOr(false, true, false) 1`] = `
"
export const andOr = (a, b, c) => a || c;"
`;

exports[`test andOr(false, true, true) 1`] = `
"
export const andOr = (a, b, c) => a || c;"
`;

exports[`test andOr(true, false, false) 1`] = `
"
export const andOr = (a, b, c) => a && b || c;"
`;

exports[`test andOr(true, false, true) 1`] = `
"
export const andOr = (a, b, c) => a && b || c;"
`;

exports[`test andOr(true, true, false) 1`] = `
"
export const andOr = (a, b) => a && b;"
`;

exports[`test andOr(true, true, true) 1`] = `
"
export const andOr = (a, b) => a && b;"
`;

exports[`test andOrAnd(false, false, false, false) 1`] = `
"
export const andOrAnd = (a, b, c) => a || c;"
`;

exports[`test andOrAnd(false, false, false, true) 1`] = `
"
export const andOrAnd = (a, b, c) => a || c;"
`;

exports[`test andOrAnd(false, false, true, false) 1`] = `
"
export const andOrAnd = (a, b, c, d) => a || c && d;"
`;

exports[`test andOrAnd(false, false, true, true) 1`] = `
"
export const andOrAnd = (a, b, c, d) => a || c && d;"
`;

exports[`test andOrAnd(false, true, false, false) 1`] = `
"
export const andOrAnd = (a, b, c) => a || c;"
`;

exports[`test andOrAnd(false, true, false, true) 1`] = `
"
export const andOrAnd = (a, b, c) => a || c;"
`;

exports[`test andOrAnd(false, true, true, false) 1`] = `
"
export const andOrAnd = (a, b, c, d) => a || c && d;"
`;

exports[`test andOrAnd(false, true, true, true) 1`] = `
"
export const andOrAnd = (a, b, c, d) => a || c && d;"
`;

exports[`test andOrAnd(true, false, false, false) 1`] = `
"
export const andOrAnd = (a, b, c) => a && b || c;"
`;

exports[`test andOrAnd(true, false, false, true) 1`] = `
"
export const andOrAnd = (a, b, c) => a && b || c;"
`;

exports[`test andOrAnd(true, false, true, false) 1`] = `
"
export const andOrAnd = (a, b, c, d) => a && b || c && d;"
`;

exports[`test andOrAnd(true, false, true, true) 1`] = `
"
export const andOrAnd = (a, b, c, d) => a && b || c && d;"
`;

exports[`test andOrAnd(true, true, false, false) 1`] = `
"
export const andOrAnd = (a, b) => a && b;"
`;

exports[`test andOrAnd(true, true, false, true) 1`] = `
"
export const andOrAnd = (a, b) => a && b;"
`;

exports[`test andOrAnd(true, true, true, false) 1`] = `
"
export const andOrAnd = (a, b) => a && b;"
`;

exports[`test andOrAnd(true, true, true, true) 1`] = `
"
export const andOrAnd = (a, b) => a && b;"
`;

exports[`test orAndOr(false, false, false, false) 1`] = `
"
export const orAndOr = (a, b) => a || b;"
`;

exports[`test orAndOr(false, false, false, true) 1`] = `
"
export const orAndOr = (a, b) => a || b;"
`;

exports[`test orAndOr(false, false, true, false) 1`] = `
"
export const orAndOr = (a, b) => a || b;"
`;

exports[`test orAndOr(false, false, true, true) 1`] = `
"
export const orAndOr = (a, b) => a || b;"
`;

exports[`test orAndOr(false, true, false, false) 1`] = `
"
export const orAndOr = (a, b, c, d) => (a || b) && (c || d);"
`;

exports[`test orAndOr(false, true, false, true) 1`] = `
"
export const orAndOr = (a, b, c, d) => (a || b) && (c || d);"
`;

exports[`test orAndOr(false, true, true, false) 1`] = `
"
export const orAndOr = (a, b, c) => (a || b) && c;"
`;

exports[`test orAndOr(false, true, true, true) 1`] = `
"
export const orAndOr = (a, b, c) => (a || b) && c;"
`;

exports[`test orAndOr(true, false, false, false) 1`] = `
"
export const orAndOr = (a, b, c, d) => a && (c || d);"
`;

exports[`test orAndOr(true, false, false, true) 1`] = `
"
export const orAndOr = (a, b, c, d) => a && (c || d);"
`;

exports[`test orAndOr(true, false, true, false) 1`] = `
"
export const orAndOr = (a, b, c) => a && c;"
`;

exports[`test orAndOr(true, false, true, true) 1`] = `
"
export const orAndOr = (a, b, c) => a && c;"
`;

exports[`test orAndOr(true, true, false, false) 1`] = `
"
export const orAndOr = (a, b, c, d) => a && (c || d);"
`;

exports[`test orAndOr(true, true, false, true) 1`] = `
"
export const orAndOr = (a, b, c, d) => a && (c || d);"
`;

exports[`test orAndOr(true, true, true, false) 1`] = `
"
export const orAndOr = (a, b, c) => a && c;"
`;

exports[`test orAndOr(true, true, true, true) 1`] = `
"
export const orAndOr = (a, b, c) => a && c;"
`;
